pyk/src/pyk/utils.py

```diff
@@ -91,6 +93,16 @@ def intersperse(iterable: Iterable[T], delimiter: T) -> Iterator[T]:
         yield x
 
 
+def unique(iterable: Iterable[H]) -> Iterator[H]:
+    elems = set()
+    for elem in iterable:
+        if elem in elems:
+            continue
+        else:
+            elems.add(elem)
+            yield elem
+
+
 def dedupe(xs: Iterable[T]) -> List[T]:
     res = []
```

Comment:
Is `dedupe` used anywehre in this library at hte moment? If not, let's remove it (and force downstream projects to switch to `unique`).

Reply:
Done.